src/OpenLoco/src/GameCommands/Road/RemoveRoad.cpp

```diff
@@ -0,0 +1,262 @@
+            {
+                continue;
+            }
+            if (elRoad->roadObjectId() != args.objectId)
+            {
+                continue;
+            }
+            if (elRoad->roadId() != args.roadId)
+            {
+                continue;
+            }
+            if (elRoad->isGhost() != ((flags & Flags::ghost) != 0))
+            {
+                continue;
+            }
+            if (elRoad->isAiAllocated() != ((flags & Flags::aiAllocated) != 0))
+            {
+                continue;
+            }
+            // Ghost only as this is checked elsewhere for non-ghost so that
+            // neutral company is always allowed
+            if (((flags & Flags::ghost) != 0) && elRoad->owner() != companyId)
+            {
+                return nullptr;
+            }
+
+            return elRoad;
+        }
+
+        return nullptr;
+    }
+
+    // 0x00477A10
+    static currency32_t roadRemoveCost(const RoadRemovalArgs& args, const World::TrackData::PreviewTrack roadPiece0, const World::Pos3 roadStart, const uint8_t flags)
+    {
+        const auto roadLoc = roadStart + World::Pos3{ Math::Vector::rotate(World::Pos2{ roadPiece0.x, roadPiece0.y }, args.rotation), roadPiece0.z };
+
+        auto tile = World::TileManager::get(roadLoc.x, roadLoc.y);
+        auto* roadElPiece = getRoadElement(tile, args, roadPiece0.index, flags);
+        if (roadElPiece == nullptr)
+        {
+            return 0;
+        }
+
+        currency32_t totalRemovalCost = 0;
+
+        const auto* roadObj = ObjectManager::get<RoadObject>(roadElPiece->roadObjectId());
+        {
+            const auto trackBaseCost = Economy::getInflationAdjustedCost(roadObj->sellCostFactor, roadObj->costIndex, 10);
+            const auto cost = (trackBaseCost * World::TrackData::getTrackMiscData(args.roadId).costFactor) / 256;
+            totalRemovalCost += cost;
+        }
+
+        // Check mod removal costs
+        for (auto i = 0U; i < 2; i++)
+        {
+            if (roadElPiece->hasMod(i))
+            {
+                const auto* trackExtraObj = ObjectManager::get<RoadExtraObject>(roadObj->mods[i]);
+                const auto trackExtraBaseCost = Economy::getInflationAdjustedCost(trackExtraObj->sellCostFactor, trackExtraObj->costIndex, 10);
+                const auto cost = (trackExtraBaseCost * World::TrackData::getTrackMiscData(args.roadId).costFactor) / 256;
+                totalRemovalCost += cost;
+            }
+        }
+
+        return totalRemovalCost;
+    }
+
+    // 0x004775A5
+    static uint32_t removeRoad(const RoadRemovalArgs& args, uint8_t flags)
+    {
+        setExpenditureType(ExpenditureType::Construction);
+        setPosition(args.pos + World::Pos3{ 16, 16, 0 });
+
+        // 0x0047762D
+        auto tile = World::TileManager::get(args.pos.x, args.pos.y);
+
+        if (auto* roadEl = getRoadElement(tile, args, args.sequenceIndex, flags); roadEl != nullptr)
+        {
```

Comment:
if roadElement is nullptr we should safely handle that and return FAILURE (this would be a change from vanilla)